### PR TITLE
build: push optimize images to infosec registry

### PR DESCRIPTION
**Pull request by misiekhardcore**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

related to https://github.com/camunda/camunda-optimize/issues/14115

### CODE REVIEW

**Comment by misiekhardcore** [comment]
@jjettenCamunda I opened this PR to investigate problems with authentication to the infosec registry.
Adding this registry was initially done in here, but it was not working so we reverted it, but it has to be fixed before we build the final 8.6 release so I need your assistance here to fix the issue. I created the  `.github/workflows/optimize-test-additional-artifactory.yml`(**Test additional artifactory**) with only all the login/auth steps from Optimize release workflow to make it easier to test.
Please take a look at the error message

**Comment by jjettenCamunda** [comment]
I think I found the issue, there is a filter on public repositories calling our GCP auth, I'll have to relax that restriction for camunda/camunda, I'll let you know once I've adjusted it

**Comment by misiekhardcore** [comment]
@jjettenCamunda it looks like it works now 💪🏽 thanks!
I will now reuse this PR to add these changes to the optimize-release-optimize-c8-only.yml` and ask you one last time to review it

File: .github/workflows/optimize-release-optimize-c8-only.yml [hunk 1]
```diff
@@ -255,12 +271,14 @@ jobs:
...
           tags=("${DOCKER_IMAGE_TEAM}:${RELEASE_VERSION}")
           tags+=("${DOCKER_IMAGE_DOCKER_HUB}:${RELEASE_VERSION}")
+          tags+=("${DOCKER_INTERNAL_IMAGE_DOCKER_HUB}:${RELEASE_VERSION}")
 
           # Major and minor versions are always tagged as the latest
           if [ "${MAJOR_OR_MINOR}" = true ] || [ "${DOCKER_LATEST}" = true ]; then
               echo "Tagging optimize release docker image with \`${DOCKER_LATEST_TAG}\`"
               tags+=("${DOCKER_IMAGE_TEAM}:${DOCKER_LATEST_TAG}")
               tags+=("${DOCKER_IMAGE_DOCKER_HUB}:${DOCKER_LATEST_TAG}")
+              tags+=("${DOCKER_INTERNAL_IMAGE_DOCKER_HUB}:${DOCKER_LATEST_TAG}")
```
Comment (jjettenCamunda) [inline .github/workflows/optimize-release-optimize-c8-only.yml:281]: No need to push latest to the internal repo, CVE-Registry is configured to ignore latest

Comment (misiekhardcore) [reply .github/workflows/optimize-release-optimize-c8-only.yml]: Okay, I deleted it

**Review by jjettenCamunda — APPROVED** [review]
LGTM

**Pull request merged**